forte/data/base_store.py

```diff
@@ -31,6 +32,81 @@ def __init__(self):
         various fields stored in entry lists.
         """
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def serialize(
+        self,
+        output_path: str,
+        serialize_method: str = "json",
+        save_attribute: bool = True,
+    ):
+        if serialize_method == "json":
+            with open(output_path, mode="wt", encoding="utf-8") as json_out:
+                json_out.write(self.to_string(serialize_method, save_attribute))
+        else:
+            raise NotImplementedError(
+                f"Unsupported serialization method {serialize_method}"
+            )
+
+    def to_string(
+        self,
+        json_method: str = "json",
+        save_attribute: bool = True,
+    ) -> str:
+        """
+        Return the string representation (json encoded) of this method.
+
+        Args:
+            json_method: What method is used to convert data pack to json.
+                Only supports `json` for now. Default value is `json`.
+            save_attribute: Boolean value indicating whether users want to
+                save `_type_attributes`.
+
+        Returns: String representation of the data pack.
+        """
+        if json_method == "json":
+            state = self.__getstate__()
+            if not save_attribute:
+                state.pop("fields")
+            return json.dumps(state, indent=2)
+        else:
+            raise ValueError(f"Unsupported JSON method {json_method}.")
+
+    @classmethod
+    def _deserialize(
```

Comment:
why is this private?

Reply:
since we also have `deserialize` in data_store.py, we might need different names in this base_store.py. I am following the format of base_pack and data_pack.